page_objects/examples/weather_shopper_mobile_app/weather_shopper_payment_page.py

```diff
@@ -97,3 +101,82 @@ def submit_payment_details(self,card_type,email,card_number,card_expiry,card_cvv
         result_flag &= self.enter_card_cvv(card_cvv)
         result_flag &= self.submit_payment()
         return result_flag
+
+    @Wrapit._exceptionHandler
+    def navigate_to_field(self, fieldname,screenshotname):
+        """
+        Navigating the cursor to the desired field
+        and capturing the screenshot of the error prompt.
+        """
+        # Define the actions for each fieldname
+        field_actions = {
+            locators.fieldnames[0]: [locators.payment_email],
+            locators.fieldnames[1]: [locators.payment_card_number],
+            locators.fieldnames[2]: [locators.payment_email, locators.payment_card_expiry],
+            locators.fieldnames[3]: [locators.payment_email, locators.payment_card_cvv]
+        }
+        actions = field_actions.get(
+            fieldname, [locators.payment_email, locators.payment_card_expiry]
+        )
+        # Perform the actions
+        for action in actions:
+            result_flag = self.click_element(action)
+
+        self.hide_keyboard()
+        self.save_screenshot(screenshot_name=screenshotname)
+
+        return result_flag
+
+    @Wrapit._exceptionHandler
+    def image_to_string(self, image_path, substring):
+        """
+        image_to_string() - The method extracts the text from the 
+        image using tesseract.
+        find_string() - Search the substring
+        preprocess_image() - To enhance the image contrast 
+        """
+        # Use glob to find the file
+        files = glob.glob(image_path)
+        if files:
+            # Assuming there is only one file matching the pattern
+            file_path = files[0]
+            print(f"Found file: {file_path}")
+
+            # Load the image
+            image = Image.open(file_path)
+            # Enhance the image before OCR
+            image = self.preprocess_image(image)
+            # Perform OCR on the enhanced image
+            text = pytesseract.image_to_string(image)
+            result_flag = self.find_string(text,substring)
+            return result_flag
+
+
+    @Wrapit._exceptionHandler
+    def find_string(self,text,substring):
+        """
+        Check if the substring is in the input string
+        """
+        if substring in text:
+            print(f'Substring:{substring}')
+            return True
+
```

Comment:
The image_to_string should be get_string_from_image which should return a string if function works fine or should return null.
This method should be called from test. We can call this as validate_string. This should take in the text and substring and do the validation. Return False if the string match fails.